conmon/conmon.c

```diff
@@ -550,6 +550,45 @@ static bool read_stdio(int fd, stdpipe_t pipe, bool *eof)
 	}
 }
 
+static void on_sigchld(G_GNUC_UNUSED int signal)
+{
+	raise (SIGUSR1);
+}
+
+static void check_child_processes(GHashTable *pid_to_handler)
+{
+	void (*cb) (GPid, int, gpointer);
+
+        for (;;) {
+          int status;
+          pid_t pid = waitpid(-1, &status, WNOHANG);
+
+          if (pid < 0 && errno == EINTR)
+            continue;
```

Comment:
it is a common pattern to loop while you get `EINTR`.  Anyway, since I am using `WNOHANG` I think it is safe to drop completely this check as waitpid never blocks.